backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/document/counterDoc/C_Order_CounterDocHandler.java

```diff
@@ -115,6 +116,13 @@ public IDocument createCounterDocument(final IDocument document)
 
 		// References (should not be required)
 		counterOrder.setSalesRep_ID(order.getSalesRep_ID());
+		if (order.isDropShip() && order.getDropShip_BPartner_ID() != 0)
```

Comment:
@adi-stefan  cc @metas-rc just to be on the safe side,
instead of
```java
order.getDropShip_BPartner_ID() != 0
```
pls use
```java
order.getDropShip_BPartner_ID() > 0
```
always!